cmd/controller/main.go

```diff
@@ -27,30 +27,32 @@ import (
 	"github.com/nuclio/errors"
 )
 
-func getNamespace(namespaceArgument string) string {
-
-	// if the namespace was passed in the arguments, use that
-	if namespaceArgument != "" {
-		return namespaceArgument
-	}
+func main() {
+	var defaultResyncIntervalStr string
 
-	// if the namespace exists in env, use that
-	if namespaceEnv := os.Getenv("NUCLIO_CONTROLLER_NAMESPACE"); namespaceEnv != "" {
-		return namespaceEnv
+	if valueFromEnv := os.Getenv("NUCLIO_CONTROLLER_FUNCTION_OPERATOR_RESYNC_INTERVAL"); valueFromEnv != "" {
+		defaultResyncIntervalStr = valueFromEnv
+	} else {
+		defaultResyncIntervalStr = common.GetEnvOrDefaultString("NUCLIO_CONTROLLER_RESYNC_INTERVAL", "10m")
 	}
 
-	// if nothing was passed, listen on all namespaces
-	return "*"
-}
-
-func main() {
 	kubeconfigPath := flag.String("kubeconfig-path", os.Getenv("KUBECONFIG"), "Path of kubeconfig file")
 	namespace := flag.String("namespace", "", "Namespace to listen on, or * for all")
 	imagePullSecrets := flag.String("image-pull-secrets", os.Getenv("NUCLIO_CONTROLLER_IMAGE_PULL_SECRETS"), "Optional secret name to use for pull")
 	platformConfigurationPath := flag.String("platform-config", "/etc/nuclio/config/platform/platform.yaml", "Path of platform configuration file")
 	platformConfigurationName := flag.String("platform-config-name", common.GetEnvOrDefaultString("NUCLIO_CONTROLLER_PLATFORM_CONFIGURATION_NAME", "nuclio-platform-config"), "Platform configuration resource name")
 	functionOperatorNumWorkersStr := flag.String("function-operator-num-workers", common.GetEnvOrDefaultString("NUCLIO_CONTROLLER_FUNCTION_OPERATOR_NUM_WORKERS", "4"), "Set number of workers for the function operator (optional)")
-	functionOperatorResyncIntervalStr := flag.String("function-operator-resync-interval", common.GetEnvOrDefaultString("NUCLIO_CONTROLLER_FUNCTION_OPERATOR_RESYNC_INTERVAL", "10m"), "Set resync interval for the function operator (optional)")
+
+	resyncIntervalStr := flag.String("resync-interval", defaultResyncIntervalStr, "Set resync interval for the function operator (optional)")
+
+	// Deprecated: resync interval is commonly used by functions and api gateways
```

Comment:
instead or changing the name. why not just fix the issue?
Have a distinct args for apigateways and functions. so they are properly and independently configurable

Reply:
I dont think there's a real need to provide such configuration granularity. pretty much KISSed it

Reply:
This harms the consistency of the configuration here in a weird way - you configure some operators individually but binded together functions and apigateways for an arbitrary reason.... sorry but I disagree with that KISSing

Reply:
for now, leaving it for when the time comes. There's nothing using it anyway.